### PR TITLE
Add token based extractor

### PR DESCRIPTION
**Pull request by ydai1124**
Added a simple token based extractor, and a simple converter. @ibuenros Can you take a look?

### CODE REVIEW

File: gobblin-core/src/main/java/gobblin/converter/string/StringSplitterToListConverter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  @Override
+  public Converter<String, String, String, List<String>> init(WorkUnitState workUnit) {
+    String stringSplitterDelimiterKey =
+        ForkOperatorUtils.getPropertyNameForBranch(workUnit, ConfigurationKeys.CONVERTER_STRING_SPLITTER_DELIMITER);
+    Preconditions.checkArgument(workUnit.contains(stringSplitterDelimiterKey),
+        "Cannot use " + this.getClass().getName() + " with out specifying "
+            + ConfigurationKeys.CONVERTER_STRING_SPLITTER_DELIMITER);
+
+    this.splitter = Splitter.on(workUnit.getProp(stringSplitterDelimiterKey));
```
Comment (ibuenros) [inline gobblin-core/src/main/java/gobblin/converter/string/StringSplitterToListConverter.java]: Should we support trims, omit empty tokens, etc.?

File: gobblin-core/src/main/java/gobblin/converter/string/StringSplitterToListConverter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  @Override
+  public String convertSchema(String inputSchema, WorkUnitState workUnit)
+      throws SchemaConversionException {
+    return inputSchema;
+  }
+
+  @Override
+  public Iterable<List<String>> convertRecord(String outputSchema, String inputRecord, WorkUnitState workUnit)
+      throws DataConversionException {
+    return new SingleRecordIterable<>(this.splitter.splitToList(inputRecord));
```
Comment (ibuenros) [inline gobblin-core/src/main/java/gobblin/converter/string/StringSplitterToListConverter.java]: Might be worth also offering an explode mode, each token becomes a record. Not necessary for this PR, but fairly easy to add here.

File: gobblin-core/src/main/java/gobblin/converter/string/StringSplitterToListConverter.java (hunk 1 above)
Comment (ydai1124) [inline gobblin-core/src/main/java/gobblin/converter/string/StringSplitterToListConverter.java]: Will do.

**Comment by ydai1124** [comment]
@ibuenros Comments addressed.

File: gobblin-core/src/test/java/gobblin/converter/string/StringSplitterToListConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.DataConversionException;
+
+import avro.shaded.com.google.common.collect.Lists;
```
Comment (ibuenros) [inline gobblin-core/src/test/java/gobblin/converter/string/StringSplitterToListConverterTest.java]: wrong import, please use `com.google...`

File: gobblin-core/src/test/java/gobblin/source/extractor/filebased/TokenizedFileDownloaderTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+package gobblin.source.extractor.filebased;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+
+import org.testng.annotations.Test;
+
+import avro.shaded.com.google.common.base.Joiner;
+import junit.framework.Assert;
```
Comment (ibuenros) [inline gobblin-core/src/test/java/gobblin/source/extractor/filebased/TokenizedFileDownloaderTest.java]: The above two imports are incorrect, should use `testng` assert, and non-shaded guava

**Pull request merged**